### PR TITLE
feat: upgrade istio-operator to version 1.16.4

### PR DESCRIPTION
**Pull request by Ani1357**
This PR is updating the istio-operator helm chart to version 1.16.4

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [x] Helm releases are meeting otomi's baseline security policies, if applicable.
- [x] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

File: charts/istio-operator/files/gen-operator.yaml [hunk 1]
```diff
@@ -175,7 +175,7 @@ spec:
       serviceAccountName: istio-operator
       containers:
         - name: istio-operator
-          image: istio/operator:1.15.5
+          image: gcr.io/istio-testing/operator:1.16-dev
```
Comment (j-zimnowoda) [inline charts/istio-operator/files/gen-operator.yaml:178]: please do not use `dev` version. Is should be `1.16.4`

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
could pleas fix the tag.

File: charts/istio-operator/files/gen-operator.yaml (hunk 1 above)
Comment (Ani1357) [reply charts/istio-operator/files/gen-operator.yaml]: I could not find the tag 1.16.4 on gcr.io
Screenshot 2023-04-24 at 16 00 57

but I tested by using the version tagged 1.16.4 on docker.io and it seems to work fine
Screenshot 2023-04-24 at 16 05 29

should we use that one?

Comment (j-zimnowoda) [reply charts/istio-operator/files/gen-operator.yaml]: yes

**Pull request merged**